aws/table_aws_route53_resolver_rule.go

```diff
@@ -0,0 +1,301 @@
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsRoute53ResolverRule")
+
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.GetResolverRuleInput{
+		ResolverRuleId: &id,
+	}
+
+	// Get call
+	data, err := svc.GetResolverRule(params)
+	if err != nil {
+		logger.Debug("getAwsRoute53ResolverRule", "ERROR", err)
+		return nil, err
+	}
+
+	return data.ResolverRule, nil
+}
+
+func listResolverRuleAssociation(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listResolverRuleAssociation")
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	resolverRuleData := h.Item.(*route53resolver.ResolverRule)
+	id := resolverRuleData.Id
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.ListResolverRuleAssociationsInput{
+		Filters: []*route53resolver.Filter{
+			{
+				Name: types.String("ResolverRuleId"),
+				Values: []*string{
+					id,
+				},
+			},
+		},
+	}
+
+	op, err := svc.ListResolverRuleAssociations(params)
+	if err != nil {
+		logger.Debug("listResolverRuleAssociation", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func getAwsRoute53ResolverRuleTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsRoute53ResolverRuleTags")
+
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	resolverRuleData := h.Item.(*route53resolver.ResolverRule)
+
+	// For default resolver rule i.e not supported tag
+	defaultID := "rslvr-autodefined-rr-internet-resolver"
+	if *resolverRuleData.Id == defaultID {
+		return nil, nil
+	}
+
+	// Create Session
+	svc, err := Route53Resolver(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &route53resolver.ListTagsForResourceInput{
+		ResourceArn: resolverRuleData.Arn,
+	}
+
+	// Get call
+	op, err := svc.ListTagsForResource(params)
+	if err != nil {
+		logger.Debug("getAwsRoute53ResolverRuleTags", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func route53resolverRuleTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("route53resolverTagListToTurbotTags")
+
+	if d.Value == nil {
```

Comment:
Why we are doing this check , is this required ? Or we should just check the `tagList`.

Reply:
this is required. Because default rule doesn,t  suppord tags.